ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -411,21 +412,10 @@ namespace Ikarus {
       DisplacementBasedElement::calculateVectorImpl(par, force, dx);
       if (isDisplacementBased()) return;
       using namespace Dune;
-      const auto& d       = par.getGlobalSolution(Ikarus::FESolutions::displacement);
-      auto strainFunction = DisplacementBasedElement::getStrainFunction(par, dx);
-      Eigen::VectorX<ScalarType> disp(localView().size());
-      const auto& numNodes = DisplacementBasedElement::numberOfNodes;
-
-      // FIXME this should not be needed in the future strainFunction should be able to hand out this vector
-      if (dx)
-        for (auto i = 0U; i < numNodes; ++i)
-          for (auto k2 = 0U; k2 < Traits::mydim; ++k2)
-            disp[i * Traits::mydim + k2] = dx.value()[i * Traits::mydim + k2]
-                                           + d[localView().index(localView().tree().child(k2).localIndex(i))[0]];
-      else
-        for (auto i = 0U; i < numNodes; ++i)
-          for (auto k2 = 0U; k2 < Traits::mydim; ++k2)
-            disp[i * Traits::mydim + k2] = d[localView().index(localView().tree().child(k2).localIndex(i))[0]];
+      auto strainFunction       = DisplacementBasedElement::getStrainFunction(par, dx);
+      const auto& dispInBlock   = DisplacementBasedElement::getDisplacementFunction(par, dx).second;
```

Comment:
This is really bad design. Y>ou compute here the function and throw it away. Can you implement above in fehelper a function that returns the local part of a solution vector --> basically stripping out this part from `getDisplacementFunctionImpl`